api.raml

```diff
@@ -14,488 +14,293 @@ securitySchemes:
 securedBy: [ AuthzBearerToken ]
 
 types:
-  Authentication: |
+  Authentication:
+    type: object
+    properties:
+      username: string
+      password: string
+    example: |
       {
-        "$schema": "http://json-schema.org/draft-03/schema",
-        "properties": {
-          "username": {
-            "required": true,
-            "type": "string"
-          },
-          "password": {
-            "required": true,
-            "type": "string"
-          }
-        },
-        "type": "object"
+        "username": "john_smith",
+        "password": "correcthorsebatterystaple"
       }
-  Error: |
+  Error:
+    type: object
+    properties:
+      name: string
+      message: string
+      code: integer
+      status: integer
+    example: |
       {
-        "$schema": "http://json-schema.org/draft-03/schema",
-        "properties": {
-          "name": {
-            "required": true,
-            "type": "string"
-          },
-          "message": {
-            "required": true,
-            "type": "string"
-          },
-          "code": {
-            "required": true,
-            "type": "integer"
-          },
-          "status": {
-            "required": true,
-            "type": "integer"
-          }
-        },
-        "type": "object"
+        "name": "Bad Request",
+        "message": "",
+        "code": 0,
+        "status": 400
       }
-  ReturnedUser: |
+  ReturnedUser:
+    description: >
+      Information on user record. Sub-objects mfa and method are included
+      except in a user list.
+    type: object
+    properties:
+      uuid?: string
+      employee_id?: string
+      first_name?: string
+      last_name?: string
+      display_name?: string
+      username?: string
+      email?: string
+      active?:
+        enum: [ yes, no ]
+      locked?:
+        enum: [ yes, no ]
+      last_login_utc?: string
+      manager_email?: string
+      spouse_email?: string
+      do_not_disclose: boolean
+      mfa?:
```

Comment:
`mfa` was marked as required in the previous version of the RAML file. Was it intentionally made optional?

Reply:
Great question. After revisiting this, I found that it is an artifact of the fact that the previous version defined a separate `ReturnedUser` and `UserList`. I combined them and used `ReturnedUser[]` in place of `UserList`. It seems that the definition of `UserList` was incorrect in that it omitted `Mfa` and `Method`. So to be correct I need to make them required in the new version.
As for `password`, it didn't change. It is not always included, so it is correct to marked as optional.